pkg/controllers/provisioning/provisioner.go

```diff
@@ -349,7 +349,6 @@ func (p *Provisioner) launchMachine(ctx context.Context, n *scheduler.NodeClaim,
 	}
 	instanceTypeRequirement, _ := lo.Find(machine.Spec.Requirements, func(req v1.NodeSelectorRequirement) bool { return req.Key == v1.LabelInstanceTypeStable })
 	logging.FromContext(ctx).With("machine", machine.Name, "requests", machine.Spec.Resources.Requests, "instance-types", instanceTypeList(instanceTypeRequirement.Values)).Infof("created machine")
-	p.cluster.NominateNodeForPod(ctx, machine.Name)
```

Comment:
Are we losing anything here by removing this? I know we have the same nomination call in the scheduling simulation, but I wonder why we did double nomination here anyway.

Reply:
It was actually useless, looking at it now. Since the Machine wouldn't exist in state when we made this call (the providerID would have had to have been resolved by the time that we got here for this to work and the machine reconciler would have had to reconcile the Machine with that providerID). We should have another scheduling simulation with the resolved machine after this call where we should nominate a pod to this machine. Realistically, having this mapping immediately is a little difficult since the state updates are decoupled.